src/components/KRadio/KRadio.vue

```diff
@@ -1,54 +1,79 @@
 <template>
   <div
-    :checked="isSelected"
     class="k-radio"
     :class="[
-      isTypeDefault ? 'k-radio-default' : `k-radio-${type}`,
       $attrs.class ? $attrs.class : '',
-      { 'disabled': isDisabled }
+      { 'disabled': isDisabled, 'radio-card': isCard || type === 'card', 'has-error': hasError, 'checked': isChecked, 'has-description': showDescription }
     ]"
   >
     <input
       :id="inputId"
-      :checked="isSelected"
+      :checked="isChecked"
       v-bind="modifiedAttrs"
-      class="k-input"
+      class="radio-input"
       :disabled="isDisabled"
       type="radio"
       @click="handleClick"
     >
 
-    <KLabel
-      v-if="isTypeDefault && hasLabel"
-      v-bind="labelAttributes"
-      class="k-radio-label"
-      :class="{ 'has-description': hasDescription }"
-      :for="inputId"
+    <div
+      v-if="!isCard && (label || $slots.default)"
+      class="radio-label-wrapper"
+      :class="{ 'has-description': showDescription }"
     >
-      <slot>{{ label }}</slot>
+      <KLabel
+        v-bind="labelAttributes"
+        class="radio-label"
+        :for="inputId"
+      >
+        <slot>{{ label }}</slot>
+
+        <template
+          v-if="hasTooltip"
+          #tooltip
+        >
+          <slot name="tooltip" />
+        </template>
+      </KLabel>
 
       <div
-        v-if="hasDescription"
-        class="k-radio-description"
+        v-if="showDescription"
+        class="radio-description"
       >
         <slot name="description">
-          {{ description }}
+          <p>{{ description }}</p>
         </slot>
       </div>
-
-      <template
-        v-if="hasTooltip"
-        #tooltip
-      >
-        <slot name="tooltip" />
-      </template>
-    </KLabel>
+    </div>
 
     <label
-      v-else-if="$slots.default"
+      v-else-if="label || $slots.default"
+      class="radio-card-wrapper radio-label-wrapper"
+      :class="{ 'has-label': label, 'has-description': showCardDescription }"
       :for="inputId"
+      :tabindex="isDisabled ? -1 : 0"
+      @keyup.space="handleClick"
     >
-      <slot name="default" />
+      <span
+        v-if="$slots.default"
+        class="card-content-wrapper"
+      >
+        <slot />
+      </span>
+      <span
+        v-if="label"
+        class="radio-label"
+      >
+        {{ label }}
+      </span>
```

Comment:
issue: these `span` tags need to be `div` since we're allowing users to slot content; otherwise they could have invalid HTML (e.g. a `div` within a `span`)

Reply:
But then we'd be placing `<div>` inside a `<label>` tag (which still can happen with slotted content, I know)... Do you think this `<label>` tag needs to be changed to `<div role="radio">`?

Reply:
I can't have a role of `radio`; it's not a radio it's a label.

The existing `div.radio-description` needs to be swapped out to a `span` as well since it's a child of `label`.

I think we leave them as `span` tags and add a note in the `Slots` documentation that you should not slot in block-level elements since they will be rendered inside of a `label` tag.

Reply:
Done